docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,721 @@
+
+```{todo}
+If you forget to declare a method as a `library` method, you will receive the
+following error message:
+```
+
+### Required `calldata`, `memory`, or `storage` annotations for reference types
+
+In CVL 2, methods block entries for internal functions must contain either `calldata`,
+`memory`, or `storage` annotations for all arguments with reference types (such
+as arrays).
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+(cvl2-wildcards)=
+### Summaries only apply to one contract by default
+
+In CVL 1, a summary in the `methods` block applied to all methods with the
+given signature.
+
+In CVL 2, summaries only apply to a single contract, unless the old behavior is
+explicitly requested by using `_` as the receiver.  If no contract is specified,
+the default is `currentContract`.
+
+Consider the following example:
+```cvl
+using C as c;
+
+methods {
+    function f(uint)   internal => NONDET;
+    function c.g(uint) internal => ALWAYS(4);
+    function h(uint)   internal => ALWAYS(1);
+    function _.h(uint) internal => NONDET;
+}
+```
+
+In this example, the internal function `currentContract.f` has a `NONDET`
+summary, `c.g` has an `ALWAYS` summary, a call to `currentContact.h` has an
+`ALWAYS` summary and a call to `h(uint)` on any other contract will use a
+`NONDET` summary.
+
+Summaries for specific contract methods (including the default
+`currentContract`) always override wildcard summaries.
+
+Wildcard entries cannot be declared `optional` or `envfree`, since these
+annotations only make sense for specific contract methods.
+
+```{warning}
+The meaning of your summarizations will change from CVL 1 to CVL 2.  In CVL 2,
+any entry without an `_` will only apply to a single contract!
+```
+
+(cvl2-returns)=
+### Requirements on `returns`
+
+In CVL 1, the `returns` clause on methods block entries was optional.
+CVL 2 has stricter requirements on the declared return types.
+
+Entries that apply to specific contracts (i.e. those that don't use the
+`_.f` {ref}`syntax <cvl2-wildcards>`) must include a `returns` clause if the
+contract method returns a value.  A specific-contract entry may only omit the
+`returns` clause if the contract method does not return a value.
+
+The Prover will report an error if the contract method's return type differs
+from the type declared in the `methods` block entry.
+
+```{todo}
+Error message
+```
+
+Wildcard entries must not declare return types, because they may apply to
+multiple methods that return different types.
+
+```{todo}
+Error message
+```
+
+If a wildcard entry has a ghost or function summary, the user must explicitly
+provide an `expect` clause to the summary.  The `expect` clause tells the
+Prover how to encode the value returned by the summary.  For example:
+
+```cvl
+methods {
+    function _.foo() internal => fooImpl() expect uint256 ALL;
+}
+```
+
+This entry will replace any call to any internal function `f()` with a call to
```

Comment:
```suggestion
This entry will replace any call to any internal function `foo()` with a call to
```
Should this example refer to function `foo` rather than `f` ?